### PR TITLE
Migrate to Playwright for UI tests.

### PR DESCRIPTION
**Pull request by brochington**
UI tests are currently using a mix of Jest, Puppeteer, and custom logic to perform mostly snapshot based tests. This PR implements the [Playwright](https://playwright.dev/) library, which offers a number of benefits, namely the ability to test across different browsers (chrome, safari, firefox).

### CODE REVIEW

**Comment by brochington** [comment]
> Thank you for your contribution and **Welcome to our Open Source Community!**
> 
> To make sure your pull request is accepted successfully, we ask all our open source contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement); having reviewed our contributor list, we require a CLA for the following people : (@brochington).
> 
> In order to sign a CLA with FINOS, [just submit a Pull Request](https://github.com/finos/open-developer-platform/edit/main/EASYCLA_CHANGEME.md) with a simple change to this file (adding an empty line, or some random text at the bottom); this will trigger the EasyCLA bot, which will post a comment to the Pull Request stating whether all PR contributors are covered by FINOS CLA; if not covered, the bot will post instructions on how to sign the CLA.
> 
> Thanks once again for your contribution. Let us work with you to make the CLA process quick, easy and efficient so we can move forward with reviewing and accepting your pull request. Feel free to email [help@finos.org](mailto:help@finos.org) for any questions.
> 
> _cc_ **@TheJuanAndOnly99** **@mcleo-d**

Just wanted to make a quick note to say that I have signed and submitted the CLA.

**Comment by TheJuanAndOnly99** [comment]
@cla-bot[bot] check

**Review by texodus — COMMENTED** [review]
Thanks for the PR!

I've cleaned this up and narrowed down the test failures here. Also the CLA bot has rendered this PR a comment apocalyptic wasteland. Closing in favor of #2192

File: packages/perspective-viewer-d3fc/test/js/area.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            htmlPage: "/tools/perspective-test/src/html/basic-test.html", // Should this be a relative or absolute path?
+            selector: "perspective-viewer",
+        });
+
+        await loadTableAsset(page, SUPERSTORE_CSV_PATH, {
+            plugin: "Y Area",
+            columns: ["Sales"],
+        });
+
+        await compareSVGContentsToSnapshot(
```
[the remark this answers is not in the document]
Comment (brochington) [reply packages/perspective-viewer-d3fc/test/js/area.spec.ts:30]: Yes! good catch.

**Pull request closed without merging**